### ISSUE TITLE
Nesting rule

### ISSUE THREAD

**Issue by m1y4** [realm/SwiftLint#870]
Hi, I recently started using Swiftlint with my project and don't understand why I am getting this warning for a XCTestCase method.

"Nesting violation: Types should be nested at most 1 level deep (nesting)"

```
class FakeCell: ConfirmSubmitTableViewCell {
            var awakeFromNibWasCalled = false
            private override func awakeFromNib() {
                super.awakeFromNib()
                awakeFromNibWasCalled = true
            }
        }
```

**Comment by jpsim** [comment]
You answered your own question...

The `FakeCell` class definition is nested in an `XCTestCase` subclass's method, so it's 2 levels deep.

```swift
// Level 0
class MyTests: XCTestCase {
  // Level 1
  func testExample() {
    // Level 2
    class FakeCell: ConfirmSubmitTableViewCell {
    }
  }
}
```

This rule is mostly in place to prevent needlessly nested public or internal APIs, since extreme nesting hurts type discoverability.

You can either locally disable the rule in your codebase (see [Disable rules in code](https://github.com/realm/SwiftLint#disable-rules-in-code)) or submit a PR to SwiftLint that excludes types declared in functions, or types with an ACL other than `public`, `open` or `internal`.

**Comment by marcelofabri** [comment]
Closing this due to lack of activity. Feel free to reopen if tou have any other questions!
